terror/terror.go

```diff
@@ -151,154 +175,39 @@ func (ec ErrClass) NewStd(code ErrCode) *Error {
-	line    int
-}
-
-// Class returns ErrClass
-func (e *Error) Class() ErrClass {
-	return e.class
-}
-
-// Code returns ErrCode
-func (e *Error) Code() ErrCode {
-	return e.code
-}
-
-// MarshalJSON implements json.Marshaler interface.
-func (e *Error) MarshalJSON() ([]byte, error) {
-	return json.Marshal(&struct {
-		Class ErrClass `json:"class"`
-		Code  ErrCode  `json:"code"`
-		Msg   string   `json:"message"`
-	}{
-		Class: e.class,
-		Code:  e.code,
-		Msg:   e.getMsg(),
-	})
-}
-
-// UnmarshalJSON implements json.Unmarshaler interface.
-func (e *Error) UnmarshalJSON(data []byte) error {
-	err := &struct {
-		Class ErrClass `json:"class"`
-		Code  ErrCode  `json:"code"`
-		Msg   string   `json:"message"`
-	}{}
-
-	if err := json.Unmarshal(data, &err); err != nil {
-		return errors.Trace(err)
-	}
-
-	e.class = err.Class
-	e.code = err.Code
-	e.message = err.Msg
-	return nil
-}
-
-// Location returns the location where the error is created,
-// implements juju/errors locationer interface.
-func (e *Error) Location() (file string, line int) {
-	return e.file, e.line
-}
-
-// Error implements error interface.
-func (e *Error) Error() string {
-	return fmt.Sprintf("[%s:%d]%s", e.class, e.code, e.getMsg())
-}
-
-func (e *Error) getMsg() string {
-	if len(e.args) > 0 {
-		return fmt.Sprintf(e.message, e.args...)
-	}
-	return e.message
-}
-
-// GenWithStack generates a new *Error with the same class and code, and a new formatted message.
-func (e *Error) GenWithStack(format string, args ...interface{}) error {
-	err := *e
-	err.message = format
-	err.args = args
-	return errors.AddStack(&err)
-}
-
-// GenWithStackByArgs generates a new *Error with the same class and code, and new arguments.
-func (e *Error) GenWithStackByArgs(args ...interface{}) error {
-	err := *e
-	err.args = args
-	return errors.AddStack(&err)
-}
-
-// FastGen generates a new *Error with the same class and code, and a new formatted message.
-// This will not call runtime.Caller to get file and line.
-func (e *Error) FastGen(format string, args ...interface{}) error {
-	err := *e
-	err.message = format
-	err.args = args
-	return errors.SuspendStack(&err)
-}
-
-// FastGen generates a new *Error with the same class and code, and a new arguments.
-// This will not call runtime.Caller to get file and line.
-func (e *Error) FastGenByArgs(args ...interface{}) error {
-	err := *e
-	err.args = args
-	return errors.SuspendStack(&err)
-}
-
-// Equal checks if err is equal to e.
-func (e *Error) Equal(err error) bool {
-	originErr := errors.Cause(err)
-	if originErr == nil {
-		return false
-	}
-
-	if error(e) == originErr {
-		return true
-	}
-	inErr, ok := originErr.(*Error)
-	return ok && e.class == inErr.class && e.code == inErr.code
-}
-
-// NotEqual checks if err is not equal to e.
-func (e *Error) NotEqual(err error) bool {
-	return !e.Equal(err)
+	return errors.Normalize(message, errors.MySQLErrorCode(int(code)), errors.RFCCodeText(fmt.Sprintf("%s:%d", errClass2Desc[ec], code)))
```

Comment:
What if `errClass2Desc[ec]` is empty?

Reply:
errClass2Desc[ec] will not be empty, because the caller has been init at error class register.